src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonPropertyInfoOfT.cs

```diff
@@ -114,7 +114,16 @@ private protected override void SetShouldSerialize(Delegate? predicate)
         internal override object? DefaultValue => default(T);
         internal override bool PropertyTypeCanBeNull => default(T) is null;
 
-        internal JsonConverter<T> TypedEffectiveConverter { get; private set; } = null!;
+        internal new JsonConverter<T> EffectiveConverter
```

Comment:
`new` rather than `override` (and making the base virtual?  I'm not very familiar with the usage, but that seems like a bug waiting to happen.

Reply:
The `new` property returns the same converter instance, but generic. I suppose I could override and use covariant return type but IIRC that's not available in ns2.0?

Reply:
Does it need to store it in that case, or can it do `return (JsonConverter<T>)base.EffectiveConverter;`?

Reply:
It's not negligible, and the property is consumed by the serialization hot path.